pkg/resourceinterpreter/default/native/reflectstatus.go

```diff
@@ -68,12 +69,22 @@ func reflectDeploymentStatus(object *unstructured.Unstructured) (*runtime.RawExt
 		return nil, fmt.Errorf("failed to convert DeploymentStatus from map[string]interface{}: %v", err)
 	}
 
+	resourceTemplateGenerationInt := int64(0)
+	resourceTemplateGenerationStr := util.GetAnnotationValue(object.GetAnnotations(), v1alpha2.ResourceTemplateGenerationAnnotationKey)
+	err = runtime.Convert_string_To_int64(&resourceTemplateGenerationStr, &resourceTemplateGenerationInt, nil)
+	if err != nil {
+		klog.Errorf("Failed to parse resource generation form %s(%s/%s) annotation: %v",
+			object.GetKind(), object.GetNamespace(), object.GetName(), err)
+		return nil, err
+	}
+
 	grabStatus := appsv1.DeploymentStatus{
 		Replicas:            deploymentStatus.Replicas,
 		UpdatedReplicas:     deploymentStatus.UpdatedReplicas,
 		ReadyReplicas:       deploymentStatus.ReadyReplicas,
 		AvailableReplicas:   deploymentStatus.AvailableReplicas,
 		UnavailableReplicas: deploymentStatus.UnavailableReplicas,
+		ObservedGeneration:  resourceTemplateGenerationInt,
```

Comment:
I don't get why not use `deploymentStatus.ObservedGeneration`. This seems so confusing. According to the previous logic, when the function `reflectDeploymentStatus` is executed, `status.observedGeneration` is already equal to `metadata.generation`.

Reply:
Users in the member cluster or the controller will modify resources. If `deploymentStatus.ObservedGeneration` in the member cluster is used, it cannot be consistent with generation in the resource template on the control plane.

Reply:
I got you, but let's take a look at a specific example.
![deploy-status](https://github.com/karmada-io/karmada/assets/89241565/3e692cec-1052-400e-b231-e87ee3b07323)

We assume that the reconciliation of `member1` will take some time. Before the deployment of `member1` reaches a consistent state, Is the deployment status of Karmada credible?

Reply:
I have a question: there doesn't seem to be a way to reflect the actual `.status.observedGeneration` of deployment in member clusters. If `status.ObservedGeneration` changes in a member cluster, a judgment is made at the time of `AggregationStatus`: only `status.ObservedGeneration` greater than or equal to the `.metadata.generation` in the resource template are counted.

Reply:
> We assume that the reconciliation of `member1` will take some time. Before the deployment of `member1` reaches a consistent state, Is the deployment status of Karmada credible?

will operation `retention` correct it? I find that [retention](https://github.com/karmada-io/karmada/blob/master/pkg/resourceinterpreter/default/native/retain.go#L39) just retain `replicas` for deployment.

Reply:
> I got you, but let's take a look at a specific example. ![deploy-status](https://private-user-images.githubusercontent.com/89241565/325884841-3e692cec-1052-400e-b231-e87ee3b07323.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQxMTc5MzEsIm5iZiI6MTcxNDExNzYzMSwicGF0aCI6Ii84OTI0MTU2NS8zMjU4ODQ4NDEtM2U2OTJjZWMtMTA1Mi00MDBlLWIyMzEtZTg3ZWUzYjA3MzIzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI2VDA3NDcxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVkMTU2M2MyYWViNjZiZTgzZDRhYWVmOTFjODFhYjk5NjFiY2Q0YTBiNjczMGE5N2QxZTQwMzkzMjc0ZTM0MzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.FOMcHbbAe2Nk_8Y33Ite-jnLFkF2trSjGccs64q_LTQ)
> 
> We assume that the reconciliation of `member1` will take some time. Before the deployment of `member1` reaches a consistent state, Is the deployment status of Karmada credible?

@whitewindmills  Karmada  Deployment will keep the old status instead of aggregating inconsistent status of member1. Why do you think he deployment status of Karmada is credible?

Reply:
> Why do you think the deployment status of Karmada is credible?

No, we do not have to guarantee it's trustworthy. What I mean is that I hope karmada can collect the state in real time instead of waiting until it reaches a consistent state.

Reply:
> What I mean is that I hope karmada can collect the state in real time instead of waiting until it reaches a consistent state.

My main objection is that this PR undermines this behavior.